Makefile

```diff
@@ -46,11 +47,12 @@ space := $(null) #
 hyphen:= -
 extended-name:=
 
-define add-extensions =
-	$(shell echo $(shell echo $(subst $(space),$(comma), $(strip $(shell echo $(subst $(comma),$(space), $(if $(findstring ui,$(EXTENSIONS)), "$(EXTENSIONS)$(comma)$(UI_DEPENDENCIES)", $(EXTENSIONS))) | tr ' ' '\n' | sort -u | tr '\n' ' '))) | cut -c2-) | tr -d " \t\n\r")
-endef
+merge-ui-extensions=$(subst $(1),$(2),$(if $(findstring ui,$(3)),$(3)$(1)$(4),$(3)))
+merged-extensions = $(call merge-ui-extensions,$(comma),$(space),$(EXTENSIONS),$(UI_DEPENDENCIES))
+filter-valid = $(foreach ext, $(merged-extensions), $(if $(findstring $(ext),$(ALL_EXTENSIONS)),$(ext)))
```

Comment:
Error message in case unknown extension used:
```
midgard@yggdrasil:/data/ssd/rsync/github.com/adodon2go/zot$ make binary-minimal EXTENSIONS=debug,debugg
Makefile:79: *** unknown extension: debugg.  Stop.
```